### PR TITLE
feat(attachments): u#3728 t#4003 prevent download files from deleted attachment

### PR DESCRIPTION
**Pull request by bameda**
<div align="center">

</div>

This PR add a new endpoint 

This PR adds a new endpoint (`GET /api/v2/projects/{project_id}/stories/{ref}/attachments/{attachment_id}/file/{filename}`) to protect attachment files. Instead of serving the files from /media/ directly, the API will offer an endpoint to download them:

- Currently, it allows us to prevent deleted attachment files from being downloaded even if the files have not been deleted.
- In the future we will be able to prevent users who have lost permissions on stories from being able to download them, even if they know the URL.

> NOTE: To maintain the consistency of the API, the name of the `file` field has been changed to `storaged_object` in the database, which also better represents the content of the field.

### HOW TO VALIDATE:
- [x] Review the code.
- [x] Tests are green.
- [x] Manual test:
  - Attach a file to an story.
  - Download the file.
  - Copy the download url. 
  - Delete the attachment.
  - Try to download again, you shouldn't be able to do it.

### CODE REVIEW

File: python/apps/taiga/src/taiga/attachments/migrations/0001_initial.py [hunk 1]
```diff
@@ -60,12 +60,12 @@ class Migration(migrations.Migration):
                     ),
                 ),
                 (
-                    "file",
+                    "storaged_object",
```
Comment (daniel-herrero) [inline python/apps/taiga/src/taiga/attachments/migrations/0001_initial.py:63]: Will we have any problem running the migrations for existing databases? I supposed that as the name remains the same Django will not consider it, but I'm not sure. I was thinking in  next.taiga.io where we want to preserve the data.

Comment (bameda) [reply python/apps/taiga/src/taiga/attachments/migrations/0001_initial.py:63]: Attachments are not in next.taiga yet. This is why I change the migration.

For existing databases in dev environments, we will have to recreate them

Comment (bameda) [reply python/apps/taiga/src/taiga/attachments/migrations/0001_initial.py:63]: Attachments are not in next.taiga.io yet. This is why I change the migration.

For existing databases in dev environments, we will have to recreate them

Comment (daniel-herrero) [reply python/apps/taiga/src/taiga/attachments/migrations/0001_initial.py:63]: Ok, I understand it :+1:

File: python/apps/taiga/src/taiga/stories/attachments/api.py [hunk 2]
```diff
@@ -117,6 +120,34 @@ async def delete_story_attachment(
...
+    name="project.story.attachments.file",
+    summary="Download the story attachment file",
+    responses=ERROR_404 | ERROR_422,
+    response_class=StreamingResponse,
+)
+async def get_story_attachment_file(
+    project_id: B64UUID,
+    ref: int,
+    attachment_id: B64UUID,
+    filename: str,
```
Comment (daniel-herrero) [inline python/apps/taiga/src/taiga/stories/attachments/api.py:139]: I guess this is part of the protection, as you're really accessing the storage object related to the file. Any filename will do.

I'm still missing a part here I do't fully understand.

Comment (daniel-herrero) [reply python/apps/taiga/src/taiga/stories/attachments/api.py:139]: I dont't know if it's intended, but I also miss the Postman's endpoints related to attachments

**Review by daniel-herrero — APPROVED** [review]
Good job! Don't forget the postman collection before merging it

**Pull request merged**